### ISSUE TITLE
Discuss issues in SSR in README (investigate mitigation)

### ISSUE THREAD

**Issue by rwjblue** [lifeart/ember-class-modifier#2]
Since element modifiers do not run at all in SSR / FastBoot mode, we should discuss in the README the caveats an app may have when using this addon. Specifically, if you use `{{class}}` those classes will not be present when the app is ran via SSR/FastBoot.

There are a few plausible ways to mitigate this, I'd love to have a discussion to help make a path forward...

**Comment by rwjblue** [comment]
FWIW, I opened a similar issue over in https://github.com/jelhan/ember-style-modifier/issues/11 to kick off the same discussion there.

**Comment by lifeart** [comment]
@rwjblue current modifier can be used in ssr mode, specifying default classes in `class` html attribute (modifier replace all node classes once it's called). but I agree it's kinda dirty hack

**Comment by rwjblue** [comment]
> @rwjblue current modifier can be used in ssr mode

No, actually it can't, since modifiers don't run in SSR.
 
> specifying default classes in class html attribute (modifier replace all node classes once it's called)

You are saying that you can use _both_ the modifier and the manual form like this:

```hbs
    <div 
        class="
            ui flat item 
            {{if this.loading ' loading'}} 
            {{if this.hasErrors ' error'}}"

        {{class 'ui flat item'
            loading=this.loading
            error=this.hasErrors
        }}>
    </div>
```

This would work because the non-modifier form would run in SSR, and the modifier form would take over when ran on the client.

**Comment by rwjblue** [comment]
I wonder why not use something like this instead of the current modifier implementation:

```hbs
<div
    class="ui flat item {{conditional-classes
        loading=this.loading
        error=this.hasErrors
    }}">
</div>
```

We could probably find a better name for `conditional-classes` in that example but whatever the name it would work properly in both SSR and client side render, and IMHO is arguably easier to reason about...

**Comment by lifeart** [comment]
@rwjblue yeah it make sense. I think in future we facing with a lot of issues related to modifier-ssr workflow.

in SSR we likely don't have states, so it may be rewritten as 

```hbs
   <div 
        class="ui flat item"
        {{class 'ui flat item'
            loading=this.loading
            error=this.hasErrors
        }}>
    </div>
```
I think need to decide, why modifiers don't support SSR. Maybe we  can support basic node mutations, supported by SimpleDOM.

**Comment by rwjblue** [comment]
The current thinking here is that anything needing access to an "element" shouldn't be ran in FastBoot/SSR. Its also quite likely that future iterations of FastBoot will not have access to _anything_ like an element (e.g. SimpleDOM elements will go away most likely, and we will be doing direct to string rendering).

**Comment by lifeart** [comment]
is it possible to have kinda reference in FastBoot mode (instead of Node) and provide into it low-level function to work with FastBoot primitives, related to this node?

like:
```ts
(node: DOMNode | DOMReference) {
   if (node._fastBootReference) {
     node.execute(function(maybeNode: SimpleDOMNode | string) { ... })
   } else { 
     node.classList = 'foo-bar';
   }
}

```
